services/FacilityService.php

```diff
@@ -0,0 +1,246 @@
+            "data"  => array($id),
+            "limit" => 1
+        ));
+    }
+
+    public function getFacilityForUser($userId)
+    {
+        return $this->get(array(
+            "where" => "WHERE USER.id = ?",
+            "data"  => array($userId),
+            "join"  => "JOIN users USER ON FAC.name = USER.facility",
+            "limit" => 1
+        ));
+    }
+
+    public function getFacilityForUserFormatted($userId)
+    {
+        $facility = $this->getFacilityForUser($userId);
+
+        if (!empty($facility)) {
+            $formatted  = "";
+            $formatted .= $facility["name"];
+            $formatted .= "\n";
+            $formatted .= $facility["street"];
+            $formatted .= "\n";
+            $formatted .= $facility["city"];
+            $formatted .= "\n";
+            $formatted .= $facility["state"];
+            $formatted .= "\n";
+            $formatted .= $facility["postal_code"];
+
+            return array("facility_address" => $formatted);
+        }
+
+        return array("facility_address" => "");
+    }
+
+    public function getFacilityForEncounter($encounterId)
+    {
+        return $this->get(array(
+            "where" => "WHERE ENC.encounter = ?",
+            "data"  => array($encounterId),
+            "join"  => "JOIN form_encounter ENC ON FAC.id = ENC.facility_id",
+            "limit" => 1
+        ));
+    }
+
+    public function update($data)
+    {
+        $sql  = " UPDATE facility SET";
+        $sql .= "     name='" . $data["name"] . "',";
+        $sql .= "     phone='" . $data["phone"] . "',";
+        $sql .= "     fax='" . $data["fax"] . "',";
+        $sql .= "     street='" . $data["street"] . "',";
+        $sql .= "     city='" . $data["city"] . "',";
+        $sql .= "     state='" . $data["state"] . "',";
+        $sql .= "     postal_code='" . $data["postal_code"] . "',";
+        $sql .= "     country_code='" . $data["country_code"] . "',";
+        $sql .= "     federal_ein='" . $data["federal_ein"] . "',";
+        $sql .= "     website='" . $data["website"] . "',";
+        $sql .= "     email='" . $data["email"] . "',";
+        $sql .= "     color='" . $data["color"] . "',";
+        $sql .= "     service_location='" . $data["service_location"] . "',";
+        $sql .= "     billing_location='" . $data["billing_location"] . "',";
+        $sql .= "     accepts_assignment='" . $data["accepts_assignment"] . "',";
+        $sql .= "     pos_code='" . $data["pos_code"] . "',";
+        $sql .= "     domain_identifier='" . $data["domain_identifier"] . "',";
+        $sql .= "     attn='" . $data["attn"] . "',";
+        $sql .= "     tax_id_type='" . $data["tax_id_type"] . "',";
+        $sql .= "     primary_business_entity='" . $data["primary_business_entity"] . "',";
+        $sql .= "     facility_npi='" . $data["facility_npi"] . "',";
+        $sql .= "     facility_code='" . $data["facility_code"] . "'";
+        $sql .= " WHERE id='" . $data["fid"] . "'";
```

Comment:
Since it makes more sense to do sql  escaping here (rather than upstream since this can turn into an api), would surround every $data with the add_escape_custom() function.